src/utils.cairo

```diff
@@ -276,4 +286,126 @@ mod tests {
         )
     }
+
+    #[test]
+    #[available_gas(1000000000)]
+    fn test_fast_pow() {
+        assert_eq!(fast_pow(2_u128, 3_u128), 8, "invalid result");
+        assert_eq!(fast_pow(3_u128, 4_u128), 81, "invalid result");
+
+        // Test with larger exponents
+        assert_eq!(fast_pow(2_u128, 10_u128), 1024, "invalid result");
+        assert_eq!(fast_pow(10_u128, 5_u128), 100000, "invalid result");
+    }
+
+    #[test]
+    fn test_u256_into_hash() {
+        let u256_value = u256 {
+            low: 0x1234567890abcdef1234567890abcdef_u128,
+            high: 0xfedcba0987654321fedcba0987654321_u128,
+        };
+
+        let result_hash = u256_value.into();
+
+        let expected_hash = Hash {
+            value: [
+                0xfedcba09,
+                0x87654321,
+                0xfedcba09,
+                0x87654321,
+                0x12345678,
+                0x90abcdef,
+                0x12345678,
+                0x90abcdef,
+            ],
+        };
+
+        assert_eq!(result_hash, expected_hash, "invalid results");
+    }
+
+    #[test]
+    fn test_shl() {
+        let value1: u32 = 3;
+        let shift1: u32 = 2;
+        let result = shl(value1, shift1);
+        assert_eq!(result, 12, "invalid result");
+
+        let value2: u32 = 5;
+        let shift2: u32 = 0;
+        let result = shl(value2, shift2);
+        assert_eq!(result, 5);
+    }
+
+    #[test]
+    fn test_shr() {
+        // Assuming T and U are u32 for simplicity
+        let x: u32 = 32;
+        let shift: u32 = 2;
+        let result = shr(x, shift);
+        assert_eq!(result, 8);
+
+        let shift: u32 = 32;
+        let result = shr(x, shift);
+        assert_eq!(result, 0);
+
+        let shift: u32 = 0;
+        let result = shr(x, shift);
+        assert_eq!(result, 32);
+    }
+
+    fn test_hash_to_u256() {
+        let hash_value = Hash {
+            value: [
+                0xfedcba09,
+                0x87654321,
+                0xfedcba09,
+                0x87654321,
+                0x12345678,
+                0x90abcdef,
+                0x12345678,
+                0x90abcdef,
+            ],
+        };
+
+        let result_u256 = hash_value.into();
+
+        let expected_u256 = u256 {
+            high: 0xfedcba0987654321fedcba0987654321_u128,
+            low: 0x1234567890abcdef1234567890abcdef_u128,
+        };
+
+        assert_eq!(result_u256, expected_u256, "invalid results");
+    }
+
+    fn test_hash_into_bytearray() {
+        let hash = Hash {
+            value: [
+                0x12345678_u32,
+                0x9abcdef0_u32,
+                0x11223344_u32,
+                0x55667788_u32,
+                0xaabbccdd_u32,
+                0xeeff0011_u32,
+                0x22334455_u32,
+                0x66778899_u32
+            ]
+        };
+
+        let byte_array: ByteArray = hash.into();
+
+        let expected_byte_array = {
```

Comment:
This is basically how the conversion is implemented :D
Can you use from_hex instead for expected value instead?

Reply:
can you explain further,

am not sure i understand this 

`from_hex`

Reply:
look at tests_utils.cairo file, there is a `from_hex` helper that takes a hex string and returns a `ByteArray`.  you should use  it for unit testing